### PR TITLE
feat(node/child_process): add execFile

### PR DESCRIPTION
**Pull request by Nautigsam**
Implements `execFile` function in node compat layer.
I introduced some new types and specifically new error types (`ChildProcessError` and `ExecFileError`), please challenge me on this if they should be defined elsewhere or if I should use other types instead.

### CODE REVIEW

File: node/child_process.ts [hunk 1]
```diff
@@ -116,8 +123,293 @@ export function spawn(
...
+  file: string,
+  options: ExecFileOptions,
+  callback: ExecFileCallback,
+): ChildProcess;
+export function execFile(
+  file: string,
+  args: string[],
+  options: ExecFileOptions,
+  callback: ExecFileCallback,
+): ChildProcess;
```
Comment (kt3k) [inline node/child_process.ts:165]: Nice overloads!

**Review by kt3k — APPROVED** [review]
LGTM.

`ChildProcessError` and `ExecFileError` look just fine to me.

File: node/child_process.ts (hunk 1 above)
Comment (Nautigsam) [reply node/child_process.ts:165]: Thanks 😄

**Review by bartlomieju — COMMENTED** [review]
@Nautigsam nice work! Is there any chance you could try to enable some of the Node native tests for this API? 

Take a look at `node/README.md` for information how to do that

**Comment by Nautigsam** [comment]
@bartlomieju sure, I will take a look at this ASAP.

**Comment by Nautigsam** [comment]
@bartlomieju I tried to add some Node tests but they required manual edits. I wonder if it is okay how I did.

**Comment by kt3k** [comment]
> I tried to add some Node tests but they required manual edits. I wonder if it is okay how I did.

That's ok. Thanks for doing it!

**Review by kt3k — APPROVED** [review]
LGTM. Thanks for your contribution! @Nautigsam

**Comment by kt3k** [comment]
> Node 16 supports passing a string as a code. Should we support it as well?

It should support string as well. Whatever node.js does is correct in std/node in principle

I created an issue for that https://github.com/denoland/deno_std/issues/1845

**Pull request merged**